src/priorityQueue.js

```diff
@@ -0,0 +1,33 @@
+'use strict'
+
+  export default class PriorityQueue {
+  constructor() {
+    this.pQueue = [];
```

Comment:
In a priority queue each object needs to contain data but also a priority weight. I would recommend adding these to the contructor, i.e. 
```
this.weight = 0
this.data = data
```
this makes the priority weight default to 0, but then you can edit it for each item/node/index.